tests/src/Unit/OgAccessEntityTestBase.php

```diff
@@ -48,13 +48,19 @@ public function setup() {
     $this->groupContentEntity->isNew()->willReturn(FALSE);
     $this->groupContentEntity->getEntityType()->willReturn($entity_type->reveal());
     $this->groupContentEntity->getEntityTypeId()->willReturn($entity_type_id);
+    // We should be returning a field and mocking the referencedEntities there, but I do not know how to do it.
```

Comment:
@mariano-dagostino can you have a look please (or ping if you get stuck)

Reply:
@DavidHernandez you could do something like this. However, I'm not really sure why would you try to mock everything if you are writing a Kernel test to test the actual hook.

```diff
diff --git a/tests/src/Unit/OgAccessEntityTestBase.php b/tests/src/Unit/OgAccessEntityTestBase.php
index 71270f5..f18fb1a 100644
--- a/tests/src/Unit/OgAccessEntityTestBase.php
+++ b/tests/src/Unit/OgAccessEntityTestBase.php
@@ -8,6 +8,7 @@ use Drupal\Core\Entity\EntityStorageInterface;
 use Drupal\Core\Entity\EntityTypeInterface;
 use Drupal\Core\Entity\EntityTypeManagerInterface;
 use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\Core\Field\EntityReferenceFieldItemList;
 use Drupal\Core\Field\FieldDefinitionInterface;
 use Drupal\Core\Field\FieldStorageDefinitionInterface;
 use Drupal\og\OgGroupAudienceHelperInterface;
@@ -42,6 +43,15 @@ class OgAccessEntityTestBase extends OgAccessTestBase {
     $entity_type->isSubclassOf(FieldableEntityInterface::class)->willReturn(TRUE);
     $entity_type->id()->willReturn($entity_type_id);
 
+    // Some mocked groups that are referenced by the content entity.
+    $groups = [
+      $this->prophesize(ContentEntityInterface::class),
+      $this->prophesize(ContentEntityInterface::class),
+    ];
+
+    $og_audience = $this->prophesize(EntityReferenceFieldItemList::class);
+    $og_audience->referencedEntities()->willReturn($groups);
+
     $this->groupContentEntity = $this->prophesize(ContentEntityInterface::class);
     $this->groupContentEntity->id()->willReturn($entity_id);
     $this->groupContentEntity->bundle()->willReturn($bundle);
@@ -49,8 +59,7 @@ class OgAccessEntityTestBase extends OgAccessTestBase {
     $this->groupContentEntity->getEntityType()->willReturn($entity_type->reveal());
     $this->groupContentEntity->getEntityTypeId()->willReturn($entity_type_id);
     // We should be returning a field and mocking the referencedEntities there, but I do not know how to do it.
-    $this->groupContentEntity->get('test')->willReturn($this->groupContentEntity);
-    $this->groupContentEntity->referencedEntities()->willReturn(['test']);
+    $this->groupContentEntity->get('og_audience')->willReturn($og_audience);
     $this->addCache($this->groupContentEntity);
 
     // If the group audience helper is asked if the group content entity has any
@@ -59,7 +68,7 @@ class OgAccessEntityTestBase extends OgAccessTestBase {
       ->willReturn(TRUE);
 
     $this->groupAudienceHelper->getAllGroupAudienceFields($entity_type_id, $bundle)
-      ->willReturn(['test' => 'test']);
+      ->willReturn(['og_audience' => 'og_audience']);
 
     // It is expected that a list of entity operation permissions is retrieved
     // from the permission manager so that the passed in permission can be

```